doc/pantry.md

```diff
@@ -0,0 +1,482 @@
+will clone that repo.
+
+```yaml
+extra-deps:
+- git: git@github.com:commercialhaskell/stack.git
+  commit: 6a86ee32e5b869a877151f74064572225e1a0398
+- git: git@github.com:snoyberg/http-client.git
+  commit: "a5f4f3"
+- hg: https://example.com/hg/repo
+  commit: da39a3ee5e6b4b0d3255bfef95601890afd80709
+```
+
+__NOTE__ It is highly recommended that you only use SHA1 values for a
+Git or Mercurial commit. Other values may work, but they are not
+officially supported, and may result in unexpected behavior (namely,
+Stack will not automatically pull to update to new versions).
+Another problem with this is that your build will not be deterministic,
+because when someone else tries to build the project they can get a
+different checkout of the package.
+
+A common practice in the Haskell world is to use "megarepos", or
+repositories with multiple packages in various subdirectories. Some
+common examples include [wai](https://github.com/yesodweb/wai/) and
+[digestive-functors](https://github.com/jaspervdj/digestive-functors). To
+support this, you may also specify `subdirs` for repositories, e.g.:
+
+```yaml
+extra-deps:
+- git: git@github.com:yesodweb/wai
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+  subdirs:
+  - auto-update
+  - wai
+```
+
+Since v1.7.1, you can specify packages from GitHub repository name using `github`:
+
+```yaml
+extra-deps:
+- github: snoyberg/http-client
+  commit: a5f4f30f01366738f913968163d856366d7e0342
+```
+
+If unspecified, `subdirs` defaults to `['.']` meaning looking for a
+package in the root of the repo.  Note that if you specify a value of
+`subdirs`, then `'.'` is _not_ included by default and needs to be
+explicitly specified if a required package is found in the top-level
+directory of the repository.
+
+Using the `stack freeze` command will add in additional information,
+including not only the Pantry tree hash, but also package metadata
+which can allow Stack to work faster by bypassing cabal file
+parses. For example, this:
+
+```yaml
+extra-deps:
+- git: git@github.com:yesodweb/wai
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+  subdirs:
+  - auto-update
+  - wai
+```
+
+Would be converted into:
+
+```yaml
+extra-deps:
+- subdir: auto-update
+  cabal-file:
+    size: 1219
+    sha256: c07b2b1a2df1199f83eef819ac9bb067567e100b60586a52f8b92fc733ae3a6d
+  name: auto-update
+  version: 0.1.2.1
+  git: git@github.com:yesodweb/wai
+  pantry-tree:
+    size: 687
+    sha256: 26377897f35ccd3890b4405d72523233717afb04d62f2d36031bf6b18dcef74f
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+- subdir: wai
+  cabal-file:
+    size: 1717
+    sha256: 7b46e7a8b121d668351fa8a684810afadf58c39276125098485203ef274fd056
+  name: wai
+  version: 3.0.2.3
+  git: git@github.com:yesodweb/wai
+  pantry-tree:
+    size: 10299
+    sha256: ce33fddab13592c847fbd7acd1859dfcbb9aeb6c212db3cee27c909fa3f3ae44
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
```

Comment:
To update this, do I have to delete the incorrect information, or will it regenerate the correct information if I just update the commit hash?

Reply:
You'd need to delete the incorrect information. I'll add a comment about that.